packages/commons/src/context/context.ts

```diff
@@ -1,13 +1,47 @@
-/* eslint-disable functional/immutable-data */
-import { zodiosContext } from "@zodios/express";
+import {
+  ZodiosRouterContextRequestHandler,
+  zodiosContext,
+} from "@zodios/express";
+import { v4 as uuidv4 } from "uuid";
 import { z } from "zod";
 import { AuthData } from "../auth/authData.js";
+import { Logger, logger } from "../logging/index.js";
+import { readCorrelationIdHeader } from "../auth/headers.js";
+
+export type AppContext = {
+  authData: AuthData;
+  messageData?: {
+    eventType: string;
+    eventVersion: number;
+    streamId: string;
+  };
+  correlationId?: string | null | undefined;
+};
+export type ZodiosContext = NonNullable<typeof zodiosCtx>;
+export type ExpressContext = NonNullable<typeof zodiosCtx.context>;
 
 export const ctx = z.object({
+  serviceName: z.string(),
   authData: AuthData,
   correlationId: z.string(),
 });
 
 export const zodiosCtx = zodiosContext(z.object({ ctx }));
-export type ZodiosContext = NonNullable<typeof zodiosCtx>;
-export type ExpressContext = NonNullable<typeof zodiosCtx.context>;
+export type ZodiosCtx = z.infer<typeof ctx>;
```

Comment:
The `AppContext` type looks unused - you already define the context model as `ctx` using zod and then infer it from there, but I think that `ZodiosCtx` is a little misleading as the model name for the app context (it models the req.ctx also before zodios comes into play). 
I propose to refactor these lines as follows:

```ts
export const AppContext = z.object({
  serviceName: z.string(),
  authData: AuthData,
  correlationId: z.string(),
});
export type AppContext = z.infer<typeof ctx>;

export const zodiosCtx = zodiosContext(z.object({ ctx: AppContext }));
export type ZodiosContext = NonNullable<typeof zodiosCtx>;
export type ExpressContext = NonNullable<typeof zodiosCtx.context>;
```